plugins/reports/counter/classes/CounterReport.inc.php

```diff
@@ -0,0 +1,301 @@
+			}
+			$filters[STATISTICS_DIMENSION_CONTEXT_ID] = $journalId;
+		}
+		return $filters;
+	}
+
+	/**
+	 * Given a Year-Month period and array of COUNTER\PerformanceCounters, create a COUNTER\Metric
+	 * @param string $period Date in Ym format
+	 * @param array $counters COUNTER\PerformanceCounter array
+	 * @return COUNTER\Metric
+	 */
+	protected function createMetricByMonth($period, $counters) {
+		$metric = array();
+		try {
+			$metric = new COUNTER\Metric(
+				// Date range for JR1 is beginning of the month to end of the month
+				new COUNTER\DateRange(
+					DateTime::createFromFormat('Ymd His', $period.'01 000000'),
+					DateTime::createFromFormat('Ymd His', $period.date('t', strtotime(substr($period, 0, 4).'-'.substr($period, 4).'-01')).' 235959')
+				),
+				'Requests',
+				$counters
+			);
+		} catch (Exception $e) {
+			$this->setError($e, COUNTER_EXCEPTION_ERROR | COUNTER_EXCEPTION_INTERNAL);
+		}
+		return $metric;
+	}
+
+	/**
+	 * Construct a Reports result containing the provided performance metrics
+	 * @param $reportItems array COUNTER\ReportItem
+	 * @return string xml
+	 */
+	function createXML($reportItems) {
+		$errors = $this->getErrors();
+		$fatal = false;
+		foreach ($errors as $error) {
+			if ($error->getCode() & COUNTER_EXCEPTION_ERROR) {
+				$fatal = true;
+			}
+		}
+		if (!$fatal) {
+			try {
+				$report = new COUNTER\Reports(
+					new COUNTER\Report(
+						String::generateUUID(),
+						$this->getRelease(),
+						$this->getCode(),
+						$this->getTitle(),
+						new COUNTER\Customer(
+							'0', // customer id is unused
+							$reportItems,
+							__('plugins.reports.counter.allCustomers')
+						),
+						new COUNTER\Vendor(
+							$this->getVendorID(),
+							$this->getVendorName(),
+							$this->getVendorContacts(),
+							$this->getVendorWebsiteUrl(),
+							$this->getVendorLogoUrl()
+						)
+					)
+				);
+			} catch (Exception $e) {
+				$this->setError($e, COUNTER_EXCEPTION_ERROR | COUNTER_EXCEPTION_INTERNAL);
+			}
+			if (isset($report)) {
+				return (string) $report;
+			}
+		}
+		return;
+	}
+
+	/**
+	 * Get the Vendor Id
+	 * @return $string
+	 */
+	function getVendorId() {
+		return $this->_getVendorComponent('id');
+	}
+
+	/**
+	 * Get the Vendor Name
+	 * @return $string
+	 */
+	function getVendorName() {
+		return $this->_getVendorComponent('name');
+	}
+
+	/**
+	 * Get the Vendor Contacts
+	 * @return array() COUNTER\Contact
+	 */
+	function getVendorContacts() {
+		return $this->_getVendorComponent('contacts');
+	}
+
+	/**
+	 * Get the Vendor Website URL
+	 * @return string
+	 */
+	function getVendorWebsiteUrl() {
+		return $this->_getVendorComponent('website');
+	}
+
+	/**
+	 * Get the Vendor Contacts
+	 * @return array COUNTER\Contact
+	 */
+	function getVendorLogoUrl() {
+		return $this->_getVendorComponent('logo');
+	}
+
+	/**
+	 * Get the Vendor Componet by key
+	 * @param $key string
+	 * @return mixed
+	 */
+	protected function _getVendorComponent($key) {
```

Comment:
Oops.  I think I should have omitted any function scope.  Just `function _getVendorComponent($key)`.